hub/hub.go

```diff
@@ -0,0 +1,133 @@
+// Hub is a package that describes the Hubs and clients functionalities and
+// structure for the "web-chat-go" usage.
+package hub
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+var wg = &sync.WaitGroup{}
+var mut = &sync.RWMutex{}
+
+type Hub struct {
+	// Map of clients that will be used to broadcast messages to all of them.
+	// This collection is used in order to harness the Go's built-in function
+	// called "delete".
+	clients    map[*Client]bool
+	clientsTCP map[*ClientTCP]bool
+	// A channel of []byte type that will be used to broadcast messages to
+	// the rest of Clients.
+	Broadcast chan []byte
+	// A channel of *Client type will be used for the registration of
+	// a Client.
+	register    chan *Client
```

Comment:
These *register channels feel awkward for the job they are doing — too much code to handle simple tasks. Let's look for better ways to do this.